src/lib.rs

```diff
@@ -314,6 +313,8 @@ impl Directory for ObjectStoreDirectory {
         );
         let path = Path::new(&path_str);
 
+        // lock so no one can write a dirty version
+        let _lock = self.atomic_rw_lock.lock().unwrap();
```

Comment:
Just wanted to make sure that this is unlocked when it is out of scope?

Reply:
yeah, this is a RAII lock guard.